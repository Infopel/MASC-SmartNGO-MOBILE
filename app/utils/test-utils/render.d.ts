export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
