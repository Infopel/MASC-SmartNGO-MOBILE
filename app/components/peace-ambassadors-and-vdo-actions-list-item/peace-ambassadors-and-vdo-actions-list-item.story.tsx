import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { PeaceAmbassadorsAndVdoActionsListItem } from "./peace-ambassadors-and-vdo-actions-list-item.component"

storiesOf("PeaceAmbassadorsAndVdoActionsListItem", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <PeaceAmbassadorsAndVdoActionsListItem style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
