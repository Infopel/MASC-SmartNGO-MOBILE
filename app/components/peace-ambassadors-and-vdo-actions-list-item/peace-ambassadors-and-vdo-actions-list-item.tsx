import { Card } from "components/card/card"
import { useIsPressed } from "components/hooks"
import { Icon } from "components/icon/icon"
import { Text } from "components/text/text"
import { HStack, IconButton, Pressable, usePropsResolution, VStack } from "native-base"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { Action } from "storage/models/action"
import { DateUtils } from "utils/DateUitls"
import { TextUtils } from "utils/TextUtils"

export interface PeaceAmbassadorsAndVdoActionsListItemProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  item: Action
}

/**
 * Describe your component here
 */
export const PeaceAmbassadorsAndVdoActionsListItem = function PeaceAmbassadorsAndVdoActionsListItem(props: PeaceAmbassadorsAndVdoActionsListItemProps) {
  const { style, item, ...rest } = props
  const { isPressed, pressableProps } = useIsPressed()

console.log("PeaceAmbassadorsAndVdoActionsListItemProps",{item})
  const { ...resolvedNavigateProps } = usePropsResolution('Button', { variant: 'ghost', ...rest, }, { isPressed })

  return (
    <Card borderLeftWidth={6} borderLeftColor='primary.500'>
      <Pressable accessibilityRole="button"  {...resolvedNavigateProps} {...pressableProps} >
        <HStack>
          <VStack flex='1' space={2}>
            <Text fontWeight={'bold'} >{item.name}</Text>
            <Text>Coordenado por: {item._raw.coordenador}</Text>
            <Text>{TextUtils.parseNumber(item._raw.nrParticipantesMulheres) + TextUtils.parseNumber(item._raw.nrParticipantesHomesn)} Membros</Text>
            <Text>{DateUtils.formatToShortDate(item._raw.dataFormacao)}</Text>
          </VStack>
          <IconButton icon={<Icon name='pencil' type='material-community' size='xl' />} accessibilityRole="imagebutton" />
        </HStack>
      </Pressable>
    </Card>)
}
