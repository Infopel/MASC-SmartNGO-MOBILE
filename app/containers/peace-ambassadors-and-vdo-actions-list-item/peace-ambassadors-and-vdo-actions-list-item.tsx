import * as React from "react"
import { StyleProp, TextStyle, View, ViewStyle } from "react-native"
import { observer } from "mobx-react-lite"
import { color, typography } from "../../theme"
import { Text } from "components/text/text"
import {PeaceAmbassadorsAndVdoActionsListItemStyles as styles} from "./peace-ambassadors-and-vdo-actions-list-item.styles"

export interface PeaceAmbassadorsAndVdoActionsListItemProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
}

/**
 * Describe your component here
 */
export const PeaceAmbassadorsAndVdoActionsListItemContainer = observer(function PeaceAmbassadorsAndVdoActionsListItem(props: PeaceAmbassadorsAndVdoActionsListItemProps) {
  const { style } = props
  const containerStyle = {
    ...styles.container,
    ...(style && typeof style === "object" ? style : {}),
  }

  return (
    <View testID="peaceAmbassadorsAndVdoActionsListItemContainer" style={containerStyle}>
      <Text testID='peaceAmbassadorsAndVdoActionsListItemContainerTestId' style={styles.text}>Hello</Text>
    </View>
  )
})
