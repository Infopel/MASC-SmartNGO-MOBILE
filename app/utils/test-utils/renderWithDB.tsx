jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  