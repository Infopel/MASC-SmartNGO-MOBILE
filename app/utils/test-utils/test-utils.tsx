import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

