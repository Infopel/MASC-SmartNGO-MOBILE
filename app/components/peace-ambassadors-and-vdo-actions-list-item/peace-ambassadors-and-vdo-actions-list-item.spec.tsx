import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { PeaceAmbassadorsAndVdoActionsListItem } from "./peace-ambassadors-and-vdo-actions-list-item"



describe("<PeaceAmbassadorsAndVdoActionsListItem>", () => {
  it("render", async function () {
    const component = <PeaceAmbassadorsAndVdoActionsListItem item={{
      coordinatorName: "Mateus", date: 1232134213, id: "sadasdsad", name: "asdfaskdn", numberOfActiveMembers: 8
    }}></PeaceAmbassadorsAndVdoActionsListItem>

    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
