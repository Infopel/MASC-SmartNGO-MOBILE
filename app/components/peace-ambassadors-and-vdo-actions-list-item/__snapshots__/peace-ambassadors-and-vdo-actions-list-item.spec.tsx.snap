// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeaceAmbassadorsAndVdoActionsListItem> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "borderBottomColor": "#fafafa",
      "borderLeftColor": "#ef8812",
      "borderLeftWidth": 6,
      "borderRadius": 6,
      "borderRightColor": "#fafafa",
      "borderTopColor": "#fafafa",
      "borderWidth": 1,
      "elevation": 2,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
>
  <View
    _icon={
      Object {
        "color": "primary.600",
        "size": "sm",
      }
    }
    _spinner={
      Object {
        "color": "primary.600",
        "focusable": false,
        "size": "sm",
      }
    }
    _stack={
      Object {
        "alignItems": "center",
        "space": "1.5",
      }
    }
    _text={
      Object {
        "color": "primary.600",
        "fontSize": "sm",
      }
    }
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 10,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "700",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          asdfaskdn
        </Text>
        <View
          dataSet={Object {}}
          style={
            Object {
              "height": 8,
            }
          }
        />
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Coordenado por: 
          Mateus
        </Text>
        <View
          dataSet={Object {}}
          style={
            Object {
              "height": 8,
            }
          }
        />
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          8
           Membros
        </Text>
        <View
          dataSet={Object {}}
          style={
            Object {
              "height": 8,
            }
          }
        />
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#171717",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          15/01/1970
        </Text>
      </View>
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <View
        accessibilityRole="imagebutton"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={Object {}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
